### PR TITLE
test(ui): fixes default values not set TCTC-4635

### PR DESCRIPTION
**Pull request by davinov**
For an unknown reason, the default value of a prop seemed never applied.
We found some issues about this in vue-class-component's repo, but non e of the solutions worked.
So I switched to `defineComponent`, which works like a charm.

### CODE REVIEW

File: ui/src/components/stepforms/widgets/FilterSimpleCondition.vue [hunk 1]
```diff
@@ -97,91 +96,161 @@ type OperatorOption = {
 
 export const DEFAULT_FILTER = { column: '', value: '', operator: 'eq' };
 
-@Component({
+const nullOperators: Readonly<OperatorOption[]> = [
```
[the remark this answers is not in the document]
Comment (davinov) [reply ui/src/components/stepforms/widgets/FilterSimpleCondition.vue:99]: What do you mean by a registry util?
In a dedicated file with constants (outside the component file)?
(If yes, I agree!)

File: ui/src/components/stepforms/widgets/FilterSimpleCondition.vue [hunk 2]
```diff
@@ -97,91 +96,161 @@ type OperatorOption = {
...
+      default: true,
+    },
+
+    columnTypes: {
+      type: Object as PropType<ColumnTypeMapping>,
+      default: () => ({}),
+    },
+
+    availableVariables: {
+      type: Array as PropType<VariablesBucket>,
```
[the remark this answers is not in the document]
Comment (davinov) [reply ui/src/components/stepforms/widgets/FilterSimpleCondition.vue:170]: Sure about this? Seems like the TS figures that out itself. Example:
image

File: ui/src/components/stepforms/widgets/FilterSimpleCondition.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (PrettyWood) [reply ui/src/components/stepforms/widgets/FilterSimpleCondition.vue:99]: yes

File: ui/src/components/stepforms/widgets/FilterSimpleCondition.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (PrettyWood) [reply ui/src/components/stepforms/widgets/FilterSimpleCondition.vue:170]: ah ok cool! It has been improved then! It used to be a problem

File: ui/src/components/stepforms/widgets/FilterSimpleCondition.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (davinov) [reply ui/src/components/stepforms/widgets/FilterSimpleCondition.vue:99]: OK! If it's OK, I prefer to do this later on, to first unblock Tucana's testswith the new version of weaverbird

File: ui/tests/unit/filter-simple-condition-widget.spec.ts [hunk 3]
```diff
@@ -412,7 +412,7 @@ describe.skip('Widget FilterSimpleCondition', () => {
     it('should not pass the variables props if hideColumnVariables is true', () => {
       const customProps = {
         availableVariables: ['test'],
-        variableDelimiters: ['test'],
+        variableDelimiters: {start:'te', end: 'st'},
```
Comment (alice-sevin) [inline ui/tests/unit/filter-simple-condition-widget.spec.ts]: Bien vu!

File: ui/src/components/stepforms/widgets/FilterSimpleCondition.vue [hunk 4]
```diff
@@ -255,7 +259,7 @@ export default defineComponent({
   created() {
     // In absence of condition, emit directly to the parent the default value
     if (isEqual(this.value, DEFAULT_FILTER)) {
-      this.$emit('input', DEFAULT_FILTER);
+      this.$emit('input', { ...DEFAULT_FILTER });
```
Comment (alice-sevin) [inline ui/src/components/stepforms/widgets/FilterSimpleCondition.vue:262]: Une raison de pas faire this.value plutôt? Ca éviterait de faire deux spread operator et la value est déjà isolée non :thinking: ?

**Review by alice-sevin — APPROVED** [review]
LGTM!

File: ui/src/components/stepforms/widgets/FilterSimpleCondition.vue (hunk 4 above)
Comment (davinov) [reply ui/src/components/stepforms/widgets/FilterSimpleCondition.vue:262]: Tu as raison, mieux vaut ne pas changer l'existant - ce n'est pas le but de cette PR. Je remet!

**Pull request merged**